### PR TITLE
add new pull e2e Jenkins file

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
https://github.com/pingcap/tidb-operator/issues/1443
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 - [**Blog**](https://www.pingcap.com/blog/)
 - [**For support, please contact PingCAP**](http://bit.ly/contact_us_via_github)
 
-[![Build Status](https://internal.pingcap.net/idc-jenkins/job/build_tidb_operator_master_kind/badge/icon)](https://internal.pingcap.net/idc-jenkins/job/build_tidb_operator_master_kind)
+[![Build Status](https://internal.pingcap.net/idc-jenkins/job/tidb-operator-master/badge/icon)](https://internal.pingcap.net/idc-jenkins/job/tidb-operator-master)
```
Comment (cofyc) [inline README.md:10]: new job for the master branch

File: ci/pull_e2e_kind.groovy [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                        withCredentials([
+                            string(credentialsId: "tp-codecov-token", variable: 'CODECOV_TOKEN')
+                        ]) {
+                            sh """#!/bin/bash
+                            set -eu
+                            echo "info: building"
+                            make build e2e-build
+                            if [ "${GIT_REF}" == "master" ]; then
+                                echo "info: run unit tests and report coverage results for master branch"
+                                make test GOFLAGS='-race' GO_COVER=y
```
Comment (Yisaer) [inline ci/pull_e2e_kind.groovy:293]: do we need to run unit test in this job?

Comment (cofyc) [reply ci/pull_e2e_kind.groovy:293]: in pull request jobs, no 
in master branch job, yes (reporting the test coverage for master branch)

Comment (Yisaer) [reply ci/pull_e2e_kind.groovy:293]: got it

**Pull request merged**